provision/README.md

```diff
@@ -1,43 +1,142 @@
+`/mapping` endpoint provision layout can be configured.
+
+In `config.toml` we can enlist array of things and channels that we want to create and make connections between them which we call provision layout.
+Metadata can be whatever suits your needs except that at least one thing needs to have `external_id` (which is populated with value from [request](#example)).
+For channels metadata `type` is reserved for `control` and `data` which we use with [Agent](https://github.com/mainflux/agent).
+
+Example below
+```
+[[things]]
+  name = "thing"
+
+  [things.metadata]
+    external_id = "xxxxxx"
+
+
+[[channels]]
+  name = "control-channel"
+
+  [channels.metadata]
+    type = "control"
+
+[[channels]]
+  name = "data-channel"
+
+  [channels.metadata]
+    type = "data"
+
+[[channels]]
+  name = "export-channel"
+
+  [channels.metadata]
+    type = "data"
+```
+
+## Authentication
+In order to create necessary entities provision service needs to authenticate against Mainflux API. To provide authentication credentials to the provision service you can pass it in an environment variable or in a config file as Mainflux user and password or as API token (that can be issued on users/keys endpoint). Additionally users or API token can be passed in Authorization header, this authentication takes precedence over others.
+* `username`, `password` - (`MF_PROVISION_USER`, `MF_PROVISION_PASSWORD` in [.env](../.env), `mf_user`, `mf_pass` in [config.toml](../docker/addons/provision/configs/config.toml))
+* API Key - (`MF_PROVISION_API_KEY` in [.env](../.env) or [config.toml](../docker/addons/provision/configs/config.toml))
+* ```Authorization: Token|ApiKey``` - request authorization header containing either users token or API key
+
+
+## Running
+
+Provision service can be run as a standalone or in docker composition as addon to the core docker composition.
+
+Standalone:
+```
+MF_PROVISION_BS_SVC_URL=http://localhost:8202/things MF_PROVISION_THINGS_LOCATION=http://localhost:8182 MF_PROVISION_USERS_LOCATION=http://localhost:8180 MF_PROVISION_CONFIG_FILE=docker/addons/provision/configs/config.toml build/mainflux-provision
+```
+
+Docker composition:
+```
+docker-compose -f docker/addons/provision/docker-compose.yml up
+```
+
+For the case that credentials or API token is passed in configuration call to endpoint is called:
+```
+curl -s -S  -X POST  http://localhost:8888/mapping  -H 'Content-Type: application/json' -d '{ "external_id" : "33:52:77:99:43", "external_key":"223334fw2" }'
+```
+
+In the case that provision service is not deployed with credential or you want to use user other than default one being set in environment (or config file)
+```
+curl -s -S  -X POST  http://localhost:8091/mapping -H "Authorization: $TOK" -H 'Content-Type: application/json' -d '{ "external_id" : "02:42:fE:65:D3:23", "external_key":"223334fw2" }'
```

Comment:
It's not clear what is `$TOK`, so maybe to use  `$USER_TOKEN` or `$API_TOKEN` or whatever that should be